src/indexable.cr

```diff
@@ -0,0 +1,469 @@
+# A container that allows accessing elements via a numeric index.
```

Comment:
It seems fair to asume index are 0..size-1 and that all indices should return a value without raising.

It should be in the docs for clarification I think